test/minikube/minikube_rolling_upgrade_test.go

```diff
@@ -31,6 +32,119 @@ func verifyAllProcessesRunning(t *testing.T, namespaceName string, adminPod stri
 	}, 30*time.Second)
 }
 
+func TestAdminReadinessProbe(t *testing.T) {
+	// TODO: remove this whenever the image tested in nuodb-helm-charts CI
+	// supports 'nuocmd check server' (singular)
+	if os.Getenv("NUODB_DEV") != "true" {
+		t.Skip("'nuocmd check server' is not supported in released versions")
+	}
+
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	// create a two-server domain and induce a failure that makes it
+	// impossible to elect a leader, causing 'nuocmd check server
+	// --check-converged' to fail
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{
+		SetValues: map[string]string{
+			"admin.replicas": "2",
+		},
+	}, 2, "")
+	adminStatefulSet := helmChartReleaseName + "-nuodb-cluster0"
+	admin0 := adminStatefulSet + "-0"
+	admin1 := adminStatefulSet + "-1"
+
+	// make sure both Admin Pods become Ready
+	testlib.AwaitPodUp(t, namespaceName, admin0, 120*time.Second)
+	testlib.AwaitPodUp(t, namespaceName, admin1, 120*time.Second)
+
+	// make sure direct invocation of readinessprobe script succeeds on both
+	// Admin processes
+	options := k8s.NewKubectlOptions("", "", namespaceName)
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", admin0, "--", "readinessprobe")
+	require.NoError(t, err, "readinessprobe failed: %s", output)
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "--", "readinessprobe")
```

Comment:
I dont think that you can reach this code. If the readiness probe fails here, it would have failed during `StartAdmin` and the test would have been aborted.

Reply:
It looks like `testlib.StartAdmin()` does explicitly check that the Pods are Ready. But this did not catch the fact that `readinessprobe` was not being used at all as Stanimir noticed, probably because the readiness probe is invoked on a schedule (30s, 15s, 15s, ...), which leaves plenty of time for stronger `nuocmd check` command to also be satisfied.

Reply:
StartAdmin executes 
```
		// first await could be pulling the image from the repo
		AwaitPodUp(t, namespaceName, adminName, 300*time.Second)
```

on every pod. This waits for the pod to be Ready. Which by definition means that the readiness probe had to succeed.

What am I missing?

Reply:
The test is supposed to verify expected behavior. Why shouldn't we verify that `readinessprobe` actually passes as expected? It adds nothing to the running time.

Reply:
I guess...